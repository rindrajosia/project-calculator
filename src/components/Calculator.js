import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import '../assets/style.css';

const Calculator = () => {
  const [operate, setOperate] = useState({ total: null, next: null, operation: null });

  const handleClick = buttonName => {
    const calc = calculate({ ...operate }, buttonName);
    setOperate({ ...calc });
  };

  const { next } = { ...operate };
  return (
    <>
      <div className="main">
        <h1 className="text">Let&apos;s do some math</h1>
        <div className="container">
          <Display
            result={next === null ? '0' : next}
          />
          <ButtonPanel clickHandler={handleClick} />
        </div>
      </div>
    </>
  );
};
export default Calculator;
