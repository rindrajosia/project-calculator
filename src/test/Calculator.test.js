import { render } from '@testing-library/react';
import React from 'react';
import userEvent from '@testing-library/user-event';
import Calculator from '../components/Calculator';
import renderer from 'react-test-renderer';

test('Check if id: calculator-page exist', () => {
  render(<Calculator />);
  const calculator = document.querySelector('#calculator-page');
  expect(calculator).not.toBeNull();
});

it('renders correctly', () => {
  const calculator = renderer.create(<Calculator />).toJSON();
  expect(calculator).toMatchSnapshot();
});
