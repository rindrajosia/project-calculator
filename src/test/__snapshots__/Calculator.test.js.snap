// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="main"
  id="calculator-page"
>
  <h1
    className="text"
  >
    Let's do some math
  </h1>
  <div
    className="container"
  >
    <div
      className="ecran"
    >
      0
    </div>
    <div>
      <div
        className="row"
        id="one"
      >
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="AC"
        >
          AC
        </button>
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="+/-"
        >
          +/-
        </button>
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="%"
        >
          %
        </button>
        <button
          className="default button standard"
          onClick={[Function]}
          type="button"
          value="รท"
        >
          รท
        </button>
      </div>
      <div
        className="row"
        id="two"
      >
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="7"
        >
          7
        </button>
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="8"
        >
          8
        </button>
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="9"
        >
          9
        </button>
        <button
          className="default button standard"
          onClick={[Function]}
          type="button"
          value="X"
        >
          X
        </button>
      </div>
      <div
        className="row"
        id="free"
      >
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="4"
        >
          4
        </button>
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="5"
        >
          5
        </button>
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="6"
        >
          6
        </button>
        <button
          className="default button standard"
          onClick={[Function]}
          type="button"
          value="-"
        >
          -
        </button>
      </div>
      <div
        className="row"
        id="four"
      >
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="1"
        >
          1
        </button>
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="2"
        >
          2
        </button>
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="3"
        >
          3
        </button>
        <button
          className="default button standard"
          onClick={[Function]}
          type="button"
          value="+"
        >
          +
        </button>
      </div>
      <div
        className="row"
        id="five"
      >
        <button
          className="custom button wide"
          onClick={[Function]}
          type="button"
          value="0"
        >
          0
        </button>
        <button
          className="custom button standard"
          onClick={[Function]}
          type="button"
          value="."
        >
          .
        </button>
        <button
          className="default button standard"
          onClick={[Function]}
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
